### ISSUE TITLE
Iterating over a combined TypedDict containing dictionnaries

### ISSUE THREAD

**Issue by Zalk0** [astral-sh/ty#3658]
### Summary

This error only appears when I type hint the combined dict.
I don't think I should be getting it in this case.

https://play.ty.dev/48e8133e-a435-409e-8a45-27d7d4223d94

```python
from typing import TypedDict


class SubDict(TypedDict):
    data: str


class Dict1(TypedDict, total=False):
    a: SubDict
    b: SubDict


class Dict2(TypedDict, total=False):
    c: SubDict
    d: SubDict
    e: dict


class Combined(Dict1, Dict2):
    pass


dict1 = Dict1(a=SubDict(data="some data"))
dict2 = Dict2(c=SubDict(data="some other data"), e={})
combined: Combined = {**dict1, **dict2}

# No error
print(combined.get("b", {}).get("data", "not here"))

for key, value in combined.items():
    if isinstance(value, dict):
        # Error
        print(key, value.get("data", "no data"))
```

I am running `ty check` with the default config.
```
error[no-matching-overload]: No overload of bound method `dict.get` matches arguments
  --> bug_ty.py:33:20
   |
33 |         print(key, value.get("data", "no data"))
   |                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
info: First overload defined here
    --> stdlib/builtins.pyi:3055:5
     |
3055 | /     @overload  # type: ignore[override]
3056 | |     def get(self, key: _KT, default: None = None, /) -> _VT | None:
3057 | |         """Return the value for key if key is in the dictionary, else default."""
     | |_________________________________________________________________________________^ First overload defined here
     |
info: Possible overloads for bound method `get`:
info:   (self, key: _KT@dict, default: None = None, /) -> _VT@dict | None
info:   (self, key: _KT@dict, default: _VT@dict, /) -> _VT@dict
info:   [_T](self, key: _KT@dict, default: _T, /) -> _VT@dict | _T

Found 1 diagnostic
```

### Version

ty 0.0.43

**Comment by sharkdp** [comment]
Thank you for reporting this.

It looks like a duplicate of https://github.com/astral-sh/ty/issues/1130
